spec/features/explore_page_spec.rb

```diff
@@ -63,83 +44,131 @@
-
-      # Sortable by view count
-      find('#courses [data-sort="views"].sort').click
-      expect(page).to have_selector('#courses [data-sort="views"].sort.desc')
-      find('#courses [data-sort="views"].sort').click
-      expect(page).to have_selector('#courses [data-sort="views"].sort.asc')
-
-      # Sortable by student count
-      find('#courses [data-sort="students"].sort').click
-      expect(page).to have_selector('#courses [data-sort="students"].sort.desc')
-      find('#courses [data-sort="students"].sort').click
-      expect(page).to have_selector('#courses [data-sort="students"].sort.asc')
+      find('#campaigns_list .table tbody tr', text: @campaign_name_a.title).click
+      expect(page).to have_current_path("/campaigns/#{@campaign_name_a.slug}/programs")
     end
   end
 
-  describe 'rows' do
-    let(:refs_tags_key) { 'feature.wikitext.revision.ref_tags' }
+  describe 'course list' do
+    before do
+      @course_active_name_a = create(
+        :course,
+        title: 'An awesome course starting with A',
+        start: Date.civil(2016, 1, 10),
+        end: Date.civil(2050, 1, 10),
+        slug: 'foo/my_awesome_course'
+      )
+      @course_active_name_z = create(
+        :course,
+        title: 'Z course starting with Z',
+        start: Date.civil(2016, 1, 10),
+        end: Date.civil(2050, 1, 10),
+        slug: 'foo/another_awesome_course'
+      )
+      @course_inactive = create(
+        :course,
+        title: 'course2',
+        slug: 'foo/course2',
+        start: Date.civil(2016, 1, 10),
+        end: Date.civil(2016, 2, 10)
+      )
+      CampaignsCourses.create(
+        course_id: @course_active_name_a.id,
+        campaign_id: Campaign.default_campaign.id
+      )
+      CampaignsCourses.create(
+        course_id: @course_active_name_z.id,
+        campaign_id: Campaign.default_campaign.id
+      )
+      CampaignsCourses.create(
+        course_id: @course_inactive.id,
+        campaign_id: Campaign.default_campaign.id
+      )
+    end
 
-    it 'allows navigation to a campaign page' do
-      visit '/explore'
-      find('#campaigns .table tbody tr:first-child').click
-      expect(page).to have_current_path("/campaigns/#{campaign.slug}/programs")
+    describe 'lists active courses of the default campaign' do
+      it 'for admins' do
+        allow_any_instance_of(ApplicationController).to receive(:current_user).and_return(admin)
+        visit '/explore'
+        expect(page).to have_content(@course_active_name_a.title)
+        expect(page).to have_content(@course_active_name_z.title)
+        expect(page).not_to have_content(@course_inactive.title)
+      end
+
+      it 'for everyone else' do
+        visit '/explore'
+        expect(page).to have_content(@course_active_name_a.title)
+        expect(page).to have_content(@course_active_name_z.title)
+        expect(page).not_to have_content(@course_inactive.title)
+      end
     end
 
-    it 'allows navigation to a course page' do
```

Comment:
Does this not work anymore?

Reply:
ah I think I removed this accidentally. Will add back.